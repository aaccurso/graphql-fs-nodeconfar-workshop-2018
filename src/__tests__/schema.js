const { graphql } = require('graphql');
const fs = require('fs');
const util = require('util');
const { ROOT_PATH } = require('../filesystem');
const schema = require('../schema');

const fsUnlink = util.promisify(fs.unlink);

describe('graphql-fs schema', () => {
  it('should query files and dirs', async () => {
    const source = `
    query filesAndDirs {
      files {
        name
        type
      }
      dirs {
        name
        type
        files {
          name
          type
        }
        dirs {
          name
          type
        }
      }
    }
    `;

    expect(await graphql({ schema, source })).toMatchSnapshot();
  });

  it('should ls root dir', async () => {
    const source = `
    query ls {
      ls {
        name
        type
        ... on Dir {
          ls {
            name
            type
          }
        }
      }
    }
    `;

    expect(await graphql({ schema, source })).toMatchSnapshot();
  });

  it('should ls Mother', async () => {
    const source = `
    query lsDirA {
      ls(dir: "Mother") {
        name
        type
        ... on Dir {
          parent
          ls {
            name
            type
          }
        }
      }
    }
    `;

    expect(await graphql({ schema, source })).toMatchSnapshot();
  });

  it('should execute writeFile mutation', async () => {
    const source = `
    mutation writeFile($name: String!, $content: String!) {
      writeFile(name: $name, content: $content) {
        name
      }
    }
    `;
    const variableValues = {
      name: 'test.txt',
      content: 'test',
    };

    expect(await graphql({ schema, source, variableValues })).toMatchSnapshot();
    expect(await fsUnlink(`${ROOT_PATH}/test.txt`)).toBeUndefined();
  });
});
