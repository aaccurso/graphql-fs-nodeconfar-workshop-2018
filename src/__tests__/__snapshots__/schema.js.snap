// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphql-fs schema should execute writeFile mutation 1`] = `
Object {
  "data": Object {
    "writeFile": Object {
      "name": "test.txt",
    },
  },
}
`;

exports[`graphql-fs schema should ls dirA 1`] = `
Object {
  "data": Object {
    "ls": Array [
      Object {
        "name": "fileA.txt",
        "type": "File",
      },
    ],
  },
}
`;

exports[`graphql-fs schema should ls root dir 1`] = `
Object {
  "data": Object {
    "ls": Array [
      Object {
        "ls": Array [
          Object {
            "name": "fileA.txt",
            "type": "File",
          },
        ],
        "name": "dirA",
        "type": "Dir",
      },
      Object {
        "name": "fileA.txt",
        "type": "File",
      },
      Object {
        "name": "fileB.txt",
        "type": "File",
      },
    ],
  },
}
`;

exports[`graphql-fs schema should query files and dirs 1`] = `
Object {
  "data": Object {
    "dirs": Array [
      Object {
        "dirs": Array [],
        "files": Array [
          Object {
            "name": "fileA.txt",
            "type": "File",
          },
        ],
        "name": "dirA",
        "type": "Dir",
      },
    ],
    "files": Array [
      Object {
        "name": "fileA.txt",
        "type": "File",
      },
      Object {
        "name": "fileB.txt",
        "type": "File",
      },
    ],
  },
}
`;
